test/e2etest/kubetest/kubetest_runner.go

```diff
@@ -0,0 +1,231 @@
+	ginkgoArgs := fmt.Sprintf("--test_args=--ginkgo.flakeAttempts=%o --ginkgo.dryRun=%t %s", args.FlakeAttempts, args.DryRun, args.GinkgoFocus)
+	cmd := exec.Command("go", "run", "hack/e2e.go", "--", "--provider=skeleton", "--deployment=local", "--test", "--check-version-skew=false", args.GinkgoParallel, ginkgoArgs, fmt.Sprintf("--dump=%s", args.LogDir))
+	cmd.Dir = config.KubernetesPath
+
+	var out bytes.Buffer
+	var stderr bytes.Buffer
+	cmd.Stdout = &out
+	if log.GetLevel() == log.DebugLevel {
+		cmd.Stderr = &stderr
+	}
+	err := cmd.Run()
+
+	if log.GetLevel() == log.DebugLevel {
+		log.Info(fmt.Printf(stderr.String()))
+	}
+
+	//	Output our results
+	if out.String() != "" {
+		e2eLogFilePath := filepath.Join(args.LogDir, "e2e.log")
+		err := ioutil.WriteFile(e2eLogFilePath, out.Bytes(), 0644)
+		if err != nil {
+			log.Error(errors.Wrapf(err, "Couldn't write file %s", e2eLogFilePath))
+		}
+	}
+
+	// kubetest run fails if one of the testcases failes, therefore the execution is still successful and no err needs to be returned
+	if err != nil {
+		log.Error(errors.Wrapf(err, "kubetest run failed"))
+	} else {
+		log.Info("kubtest test run successful")
+	}
+}
+
+func escapeAndConcat(descFilePath string) (concatenatedParallelTests, concatenatedSerialTests string) {
+	var serialTestsBuffer bytes.Buffer
+	var parallelTestsBuffer bytes.Buffer
+
+	file, err := os.Open(descFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+
+	if config.GinkgoParallel {
+		for scanner.Scan() {
+			testcase := strings.TrimSpace(scanner.Text())
+			if testcase == "" {
+				continue
+			}
+			if strings.Contains(testcase, "[Serial]") {
+				if serialTestsBuffer.Len() > 0 {
+					serialTestsBuffer.WriteString("|")
+				}
+				serialTestsBuffer.WriteString(escapeForRegex(testcase))
+			} else {
+				if parallelTestsBuffer.Len() > 0 {
+					parallelTestsBuffer.WriteString("|")
+				}
+				parallelTestsBuffer.WriteString(escapeForRegex(testcase))
+			}
+		}
+	} else {
+		for scanner.Scan() {
+			testcase := strings.TrimSpace(scanner.Text())
+			if testcase == "" {
+				continue
+			}
+			if serialTestsBuffer.Len() > 0 {
+				serialTestsBuffer.WriteString("|")
+			}
+			serialTestsBuffer.WriteString(escapeForRegex(testcase))
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		log.Fatal(errors.Wrapf(err, "Scanning %s failed", descFilePath))
+	}
+	return parallelTestsBuffer.String(), serialTestsBuffer.String()
+}
+
+func escapeForRegex(input string) string {
+	output := strings.Replace(regexp.QuoteMeta(input), " ", "\\s", -1)
+	return output
+}
+
+func getLinesCount(filepath string) int {
+	r, err := os.Open(filepath)
+	if err != nil {
+		log.Fatal(err)
+	}
+	buf := make([]byte, 32*1024)
+	count := 0
+	lineSep := []byte{'\n'}
+
+	for {
+		c, err := r.Read(buf)
+		count += bytes.Count(buf[:c], lineSep)
+
+		switch {
+		case err == io.EOF:
+			return count
+
+		case err != nil:
+			log.Warn(err)
+			return count
```

Comment:
shoudln't we return err if we cannot determine the num of lines. Or maybe try again?